salesforce/bulk.go

```diff
@@ -118,87 +124,20 @@ func (c *Connector) BulkWrite( //nolint:funlen,cyclop
 		return nil, fmt.Errorf("%w: %s", ErrUnsupportedMode, config.Mode)
 	}
 
-	// cretes batch upload job, returns json with id and other info
-	res, err := c.createJob(ctx, config)
```

Comment:
this portion is reused for many operations, so moved to `BulkOperation` function